card/lib/card/format/render.rb

```diff
@@ -3,27 +3,38 @@ class Format
     # View rendering methods.
     #
     module Render
+      include Render::Cache
+
       DEPRECATED_VIEWS = {
         view: :open, card: :open, line: :closed, bare: :core, naked: :core
       }.freeze
 
       def render view, args={}
-        view = canonicalize_view view
-        return if hidden_view? view, args
-        view = ok_view view, args
+        return unless (view = renderable_view(view, args))
+        args = default_render_args view, args
+        cache_render view, args do
+          render! view, args
+        end
+      rescue => e
+        rescue_view e, view
+      end
+
+      def render! view, args
```

Comment:
if we find another name for this we could use `render!` instead of `_render` for skipping permissions